.github/workflows/ci.yml

```diff
@@ -1,6 +1,10 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
Does this get fired for every push to a pull request or just when one's created?

Reply:
Both! Here's a snippet of the default `pull_request` behavior from [these docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request):

> By default, a workflow only runs when a pull_request event's activity type is `opened`, `synchronize`, or `reopened`.

As the name implies, the `opened` activity type is for whenever a PR is opened.

Here's a description of `synchronize` from [these docs](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request):

> Triggered when a pull request's head branch is updated. For example, when the head branch is updated from the base branch, when new commits are pushed to the head branch, or when the base branch is changed.